lua/ttt2/extensions/math.lua

```diff
@@ -19,3 +19,48 @@ function math.ExponentialDecay(halflife, dt)
     -- ln(0.5) = -0.69..
     return exp((-0.69314718 / halflife) * dt)
 end
+
+---
+-- Gets the index of an item in the provided table, weighted according to the weights (derived from get_weight).
+-- @param table tbl The array of items to find a weighted item in.
+-- @param function get_weight Called as get_weight(item, index). Must return number.
+-- @return number
+-- @realm shared
+function math.random_weighted(tbl, get_weight)
```

Comment:
Please keep in line with (our) lua style which would be `math.RandomWeighted` or `math.GetRandomWeighted`

in general we do not use underscores for functions and variables

Reply:
Hm, I don't know why I did that.

Reply:
On this note, I feel like it would probably be beneficial to include this in linting (if possible). I'm pretty sure `.editorconfig` can describe these rules, but I don't know if any of the Lua tooling available understands that.

Reply:
@Histalek Is our expert on that matter

Reply:
this would certainly be nice to have. Sadly i'm not aware of any lua linter checking for function names out of the box